pkgs/development/python-modules/phik/default.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  isPy3k,
+  nbconvert,
+  pytest-pylint,
+  numba,
+  pandas,
+  matplotlib
+}:
+
+buildPythonPackage rec {
+  pname = "phik";
+  version = "0.9.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    format = "wheel";
+    python = "py3";
+    abi = "none";
+    platform = "any";
+    sha256 = "c398452c5c1eea153905666b289c6a153712cf3d58811fa41e2bbbd27a65d678";
+  };
+
+  propagatedBuildInputs = [
+    nbconvert
+    pytest-pylint
+    numba
+    pandas
+    matplotlib
+  ];
+
+  # specify unpackPhase here to fix error when trying to unpack wheel file
```

Comment:
I think specifying `format = "wheel";` as one of the attributes passed to `buildPythonPackage` (in addition to `fetchPypi`) fixes the problem you've tried to solve here.

Reply:
correct

Reply:
Maybe I am missing something here. I thought so as well that I don't need to explicitly specify the `unpackPhase` when specifying `format = "wheel";.` However, I encountered the following error when building:
`do not know how to unpack source archive /nix/store/811wrrzyr3330b36zkpf42z44q9xpngr-phik-0.9.8-py3-none-any.whl`
And then building failed with exit code 1.

Reply:
I get that error too if I simply remove your `unpackPhase` and `phases`, but if I also add `format = "wheel";` it builds fine. Could you double check? To be precise, `nix-build -A python3Packages.phik` succeeds when building from your most recent commit `d515038` with the following edit:

```
diff --git a/pkgs/development/python-modules/phik/default.nix b/pkgs/development/python-modules/phik/default.nix
index 0a457a6464b..eb30929fa8c 100644
--- a/pkgs/development/python-modules/phik/default.nix
+++ b/pkgs/development/python-modules/phik/default.nix
@@ -13,6 +13,7 @@
 buildPythonPackage rec {
   pname = "phik";
   version = "0.9.8";
+  format = "wheel";
 
   src = fetchPypi {
     inherit pname version;
@@ -31,14 +32,6 @@ buildPythonPackage rec {
     matplotlib
   ];
 
-  # specify unpackPhase here to fix error when trying to unpack wheel file
-  unpackPhase = ''
-    mkdir -p dist
-    cp "$src" "dist/$(stripHash "$src")"
-  '';
-
-  phases = [ "unpackPhase" "installPhase" ];
-
   meta = with lib; {
     description = "Phi_K is a new and practical correlation coefficient based on several refinements to Pearson’s hypothesis test of independence of two variables.";
     homepage = https://phik.readthedocs.io/en/latest/;
```

Reply:
Thank you @acairncross for checking as well. I know understand what you mean by adding `format = "wheel". Completely missed that part there. Silly me. Of course it works now. :)
Should I squash everything into one single commit?

Reply:
No problem!

I don't think it's strictly necessary, but yes I would have thought it would be better to squash them.